hudi-utilities/src/test/java/org/apache/hudi/utilities/TestHoodieMetadataTableValidator.java

```diff
@@ -139,6 +139,50 @@ public void testMetadataTableValidation(String viewStorageTypeForFSListing, Stri
     assertTrue(validator.getThrowables().isEmpty());
   }
 
+  @Test
+  public void testPartitionStatsValidation() {
+    // TODO: Add validation for compaction and clustering cases
```

Comment:
you can add it with a relaxed validation - partition stats range is same or wider than that after aggregating stats for all files

Reply:
I can probably address this in HUDI-8208. Added a note in the jira.